gemrb/core/System/OS4Helpers.cpp

```diff
@@ -0,0 +1,729 @@
+
+					for (i = 0; i < len; i++) PUTWC(hsrc[i]);
+					break;
+				case 'S':
+				case 's':
+					if (half) {
+						hsrc = va_arg(ap, const char *);
+						if (hsrc == NULL)
+							hsrc = "(null)";
+
+						len = strlen(hsrc);
+
+						if (limit > 0 && len > limit)
+							len = limit;
+
+						if (width > len)
+							width -= len;
+						else
+							width = 0;
+
+						if (left == 0) {
+							for (i = 0; i < width; i++) PUTWC(' ');
+						}
+
+						for (i = 0; i < len; i++) PUTWC(hsrc[i]);
+
+						if (left != 0) {
+							for (i = 0; i < width; i++) PUTWC(' ');
+						}
+					} else {
+						src = va_arg(ap, const wchar_t *);
+						if (src == NULL)
+							src = L"(null)";
+
+						len = wcslen(src);
+
+						if (limit > 0 && len > limit)
+							len = limit;
+
+						if (width > len)
+							width -= len;
+						else
+							width = 0;
+
+						if (left == 0) {
+							for (i = 0; i < width; i++) PUTWC(' ');
+						}
+
+						for (i = 0; i < len; i++) PUTWC(src[i]);
+
+						if (left != 0) {
+							for (i = 0; i < width; i++) PUTWC(' ');
+						}
+					}
+					break;
+				case 'F':
+				case 'f':
+					prec = limit;
+
+					if (prec == -1) {
+						prec = 6;
+					}
+
+					fpval.d = va_arg(ap, double);
+
+					if (isinf(fpval.d)) {
+						if (fpval.d < 0) PUTWC('-');
+
+						if (wc == 'F') {
+							PUTWC('I');
+							PUTWC('N');
+							PUTWC('F');
+						} else {
+							PUTWC('i');
+							PUTWC('n');
+							PUTWC('f');
+						}
+
+						break;
+					}
+
+					if (isnan(fpval.d)) {
+						if (wc == 'F') {
+							PUTWC('N');
+							PUTWC('A');
+							PUTWC('N');
+						} else {
+							PUTWC('n');
+							PUTWC('a');
+							PUTWC('n');
+						}
+
+						break;
+					}
+
+					if (fpval.i[0] & 0x80000000UL) {
+						fpval.d = -fpval.d;
+						PUTWC('-');
+					}
+
+					cp = _dtoa_r(r, fpval.d, 3, prec, &expt, &dsgn, &rve);
+
+					bp = cp + prec;
+
+					if (*cp == '0' && fpval.d)
+						expt = -prec + 1;
+					bp += expt;
+
+					if (fpval.d == 0)
+						rve = bp;
+
+					while (rve < bp)
+						*rve++ = '0';
+
+					ndig = rve - cp;
+
+					if (fpval.d == 0) {
+						PUTWC('0');
+						if ((size_t)expt < ndig || alternate) {
+							PUTWC('.');
+							for (i = 0; i < (ndig - 1); i++) PUTWC('0');
+						}
+					} else if (expt <= 0) {
+						PUTWC('0');
+						if (expt || ndig) {
+							PUTWC('.');
+							for (i = 0; i < -(size_t)expt; i++) PUTWC('0');
+							for (i = 0; i < ndig; i++) PUTWC(cp[i]);
+						}
+					} else if ((size_t)expt >= ndig) {
+						for (i = 0; i < ndig; i++) PUTWC(cp[i]);
+						for (i = 0; i < ((size_t)expt - ndig); i++) PUTWC('0');
+						if (alternate) PUTWC('.');
+					} else {
+						for (i = 0; i < (size_t)expt; i++) PUTWC(cp[i]);
+						PUTWC('.');
+						for (i = expt; i < ndig; i++) PUTWC(cp[i]);
+					}
+
+					break;
+			}
+		} else {
+			PUTWC(wc);
+		}
+	}
+
+	return count;
+}
+
+
+
+int fnmatch (const char *pattern, const char *string, int flags)
```

Comment:
I wouldn't bother with this. I dropped the `fnmatch` stuff in our soon-to-be-master subviews branch. I can just back port the changes, or you can rebase your work against that branch (which would probably actually be the wisest thing)

Reply:
i can wait. No problem but however i could remove it later

Reply:
that fine, but that makes it very likely that the Amigia build will be broken in the future 🤷